flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java

```diff
@@ -1233,6 +1239,9 @@
 	/** ZooKeeper default leader port. */
 	public static final int DEFAULT_ZOOKEEPER_LEADER_PORT = 3888;
 
+	/** Defaults for ZK client security **/
+	public static final boolean DEFAULT_ZOOKEEPER_SASL_DISABLE = true;
```

Comment:
I agree but it can be argued both ways. We could keep the default to false (enable SASL client auth if not disabled explicitly through configuration file) or expect an explicit ask to enable SASL through the configuration settings. I chose later since secure ZK is not a common deployment (mostly) and moreover we also have introduced new security configurations to enable security and one could configure/adjust ZK configuration at that time.